### ISSUE TITLE
First virus mission doesn't trigger the right message to your commander

### ISSUE THREAD

**Issue by sunfall** [akarnokd/open-ig#964]
[This is my first time playing IG, so let me know if this is wrong.]

Apparently, all the FAQs online say that as soon as New Caroline gets infected with a virus for the first time, you should contact your commander and let them know.  However, instead of that being available, an "ask for reinforcements" message is triggered instead.  I played quite a bit past this, and it seemed to cause the game to skip the next mission entirely (the escort mission); I didn't actually get the option to tell my commander until, in the *second* virus triggering, I intentionally let a ship through.  Then the planetary leader complained that I had been lax, and it finally triggered my ability to contact my commander to talk about the viruses.

Now: was the original behavior a bug, or is this an unintentional change, or am I just confused? :)

**Comment by akarnokd** [comment]
Based on what they say in the hungarian cutscene, I got the impression you need two infections. The FAQ could be an oversight in the original, one of many.

The mission skip, however, sounds like a bug.

**Comment by sunfall** [comment]
I would have expected to be able to ask for help once the second infection hit rather than having to deliberately let a ship through.

**Comment by akarnokd** [comment]
You are not supposed to let a second planet get infected. If you handle the first infection, soon there will be a second infection on the same planet. This time, you can report the infection to the commander. If you don't report, you get 5 infections in total and the mission forces you to contact the commander.

**Comment by sunfall** [comment]
Right.  It never *was* a second planet getting infected; it's the first planet the second time.  I couldn't report to the commander until I let a ship through my blockade and the planetary leader complained about me not doing my job.

**Comment by law7441** [comment]
you got cure the virus 2nd time on the planet without another infection planets, before you can report to your commander. 2nd time as be sorted before it let contact your commander, it how it always been on that mission.

**Comment by sunfall** [comment]
As mentioned above, the problem wasn't that it took the second infection to get the report option; the problem was that it took me intentionally failing to quarantine the planet the second time around to get the report option, and I never had the "escort" mission that happens in between the two infections.

**Comment by mansam** [comment]
I can confirm this is happening as well on the most recent version. The first infection mission starts and then resolves within a day or two, and then it immediately starts the next infection mission. There's no option to contact the commander about the infection.

**Comment by Gosu4095** [comment]
Playing through the campaign right now.

Second infection comes and goes. 4 messages from New Caroline by this point.
Now, report repeated virus infections becomes option in send message.
Virus carriers come and are destroyed. Do not use autocombat for this.
Report virus carriers destroyed becomes option in send message, unless autocombat was used.
Go to the bar and talk to Dr. Reinhart about your vision.
Escort Admiral Benson.

To which "escort" mission that happens in between the two infections are you referring?
Is the above order of events inconsistent with the original?

**Comment by mansam** [comment]
The order of events is inconsistent. For instance, Admiral Benson's escort should take place between virus infections.

**Comment by mansam** [comment]
I'm going to jump in and see if I can sort out what's going on with the mission scripting for this.

**Comment by akarnokd** [comment]
Thanks @mansam!

**Comment by Blakeley00** [comment]
I'm currently playing IG1 along side OpenIG just for fun and am noticing some interesting differences including what has been discussed here.

Original IG1:
- I got the first virus alert
- Straight away I was given the option to report it to the Colonel
- Turned away every ship and waited until virus cured
- Satellites arrived
- I detected the virus ship & destroyed it
- Was able to completely avoid a second infection
- I got the escort Admiral mission.

OpenIG:
- I got the first virus alert
- Was given NO option to report to Colonel
- Turned away every ship and waited until virus cured
- Few days later I got the second virus alert
- Was STILL given NO option to report to Colonel
- Turned away every ship and waited until virus cured
- STILL no message option, got worried & found this issue thread
- Reloaded a savegame during the 2nd virus outbreak
- Tried @sunfall 's trick of DELIBERATELY letting a trader land on New Caroline
- Straight away I got a message saying the virus was cured (much earlier than last time)
- Message Colonel option appeared straight away as well
- Satellites arrived
- I detected the virus ship & destroyed it
- I got the escort Admiral mission.

**Comment by akarnokd** [comment]
> Straight away I was given the option to report it to the Colonel

From a story perspective, it doesn't make sense to report **repeated** virus attack when there was only one. I believe this was an error in the original or a late change after the FMVs were recorded.

> Tried @sunfall 's trick of DELIBERATELY letting a trader land on New Caroline

One of the traders claims he has a cure, there is a 50% that letting him through ends the virus infection early. This was added in and is deliberate in Open-IG.

**Comment by Blakeley00** [comment]
Ah no worries, that explains that then. Thank you so much for a great remake. I LOVE your work!!!
